rust/backend/src/configuration.rs

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: 2024 Benedikt Zinn <benedikt.wh.zinn@gmail.com>
+//
+// SPDX-License-Identifier: MIT
+
+use shared::config::{Configuration as ProtoConfig, EbpfEntry as ProtoEbpfEntry};
+use std::fs::File;
+use std::io;
+use std::io::{BufReader, BufWriter};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct Configuration {
+    entries: Vec<EbpfEntry>,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
+struct EbpfEntry {
+    hr_name: String,            // human-readable name, for display on the frontend
+    description: String,        // human-readable description of the probe
+    ebpf_name: String,          // name of the ebpf function to actually load
+    fn_id: u64,                 // id of the daemon function to handle
```

Comment:
You can write 

```rust
/// human-readable name, for display on the frontend
hr_name: String,
/// human-readable description of the probe
description: String, 
/// name of the ebpf function to actually load      
ebpf_name: String, 
/// id of the daemon function to handle         
fn_id: u64,
```

Then the comments are docstrings that automatically show up in the generated documentation (also for language server).